### DISCUSSION TITLE
9.16对于 x-component-props 的更新会影响参数的传递

### DISCUSSION THREAD

**Discussion by yhjyang1** [alibaba/formily#2159] [Q&A 2.x] [answered]
![image](https://user-images.githubusercontent.com/37994732/133730493-ac5c908a-49fc-4dae-a353-af68d78a75a8.png)
在9.16的一个commit中，为什么给 传下去的 x-component-props 用 toJS 转换了下，这会导致下面问题

```
<SchemaField.String
        name="Test"
        x-component="Component"
        x-component-props={{
         params: { status: 'running' },
        }}
/>


const Component = ({params}) => {
  useEffect(() => {
    console.log(params)
    // 每次field的onChange都会进到这个effect，这不是期望中的
  }, [params])

};
```

重现步骤

https://codesandbox.io/s/stupefied-firefly-e9jod?file=/App.tsx

**Comment by janryWang** [comment, marked as answer]
合理行为，因为需要整体响应变化，比如你修改component[1].a.b.c，如果之前的版本是不会触发组件重新渲染的，那是BUG

**Reply by yhjyang1** [reply]
重新渲染没问题，但是为什么我例子中的params 值不变的情况下，每次在重新render的时候要把那个params改成一个新的对象呢

**Reply by janryWang** [reply]
toJS就是为了让深度属性响应变化才做的呀
